platform/security/certificate/security-crl-generator/src/main/java/org/codice/ddf/security/crl/generator/CrlGenerator.java

```diff
@@ -0,0 +1,205 @@
+
+            String localCrlPath;
+
+            if (Strings.fromByteArray(buffer).contains("-----BEGIN")) {
+              localCrlPath = crlFileLocation + PEM_CRL;
+              crlFile = new File(crlFileLocation + PEM_CRL);
+            } else {
+              localCrlPath = crlFileLocation + DEM_CRL;
+              crlFile = new File(localCrlPath);
+            }
+
+            OutputStream outStream = new FileOutputStream(crlFile);
+            outStream.write(buffer);
+            SecurityLogger.audit(
+                "Copied the content of the CRl at {} to the local CRL at {}.",
+                crlLocationUrl,
+                localCrlPath);
+
+            setCrlFileLocationInPropertiesFile(localCrlPath);
+          }
+        }
+      } catch (IOException e) {
+        LOGGER.warn("Unable to copy CRL to local CRL.");
+      }
+    } else {
+      LOGGER.warn("No/bad response");
+    }
+  }
+
+  /** Sets the org.apache.ws.security.crypto.merlin.x509crl.file property */
+  @VisibleForTesting
+  void setCrlFileLocationInPropertiesFile(String localCrlPath) {
+    addProperty(issuerSignaturePropertiesLocation, localCrlPath);
+    addProperty(issuerEncryptionPropertiesLocation, localCrlPath);
+    addProperty(serverSignaturePropertiesLocation, localCrlPath);
+    addProperty(serverEncryptionPropertiesLocation, localCrlPath);
+
+    SecurityLogger.audit(
+        "Setting the {} property to {} as signature and encryption properties.",
+        CRL_PROPERTY_KEY,
+        localCrlPath);
+  }
+
+  private void addProperty(String path, String localCrlPath) {
+    Properties properties = PROPERTIES_LOADER.loadProperties(path);
+    properties.put(CRL_PROPERTY_KEY, localCrlPath);
+    try (OutputStream outputStream = new FileOutputStream(path)) {
+      properties.store(outputStream, null);
+    } catch (IOException e) {
+      LOGGER.warn("Unable to add the {} property to the property file {}.", CRL_PROPERTY_KEY, path);
+    }
+  }
+
+  /** Removes the org.apache.ws.security.crypto.merlin.x509crl.file property */
+  void removeCrlFileLocationInPropertiesFile() {
+    removeProperty(issuerSignaturePropertiesLocation);
+    removeProperty(issuerEncryptionPropertiesLocation);
+    removeProperty(serverSignaturePropertiesLocation);
+    removeProperty(serverEncryptionPropertiesLocation);
+
+    SecurityLogger.audit(
+        "Removing the {} property from signature and encryption properties.", CRL_PROPERTY_KEY);
+  }
+
+  private void removeProperty(String path) {
+    Properties properties = PROPERTIES_LOADER.loadProperties(path);
+    properties.remove(CRL_PROPERTY_KEY);
+    try (OutputStream outputStream = new FileOutputStream(path)) {
+      properties.store(outputStream, null);
+    } catch (IOException e) {
+      LOGGER.warn(
+          "Unable to remove the {} property to the property file {}.", CRL_PROPERTY_KEY, path);
+    }
+  }
+
+  public void setCrlLocationUrl(String crlLocationUrl) {
+    this.crlLocationUrl = crlLocationUrl;
```

Comment:
❗️  It seems weird to me that we rely on the call to this setter by blueprint whenever to execute the download and save. 
You are blocking blueprint which might not be a good idea to do. It might better to register a configuration listener with config admin and detect the change.
❓ What provision do we have to do this download if the configuration is changed when DDF is offline?
❓ How do we handle cases where DDF is shutdown before we finish the download?